ProcessOptimizer/space/constraints.py

```diff
@@ -476,6 +630,55 @@ def __eq__(self, other):
         else:
             return False
 
+class SumEquals():
+    def __init__(self, dimensions, value):
+        """Constraint class of type SumEquals.
+
+        This constraint enforces that the sum of all values drawn for the
+        specified dimensions, is exactly equal to a value. Can only be
+        used with real dimensions.
+
+        Parameters
+        ----------
+        * `dimensions` [list of ints]:
+            A list of integers coresponding to the index of the dimensions that
+            should be summed
+
+        * `value` [float or int]:
+            The value for which the sum should be equal to.
+
+        """
+
+        if not (type(dimensions) == tuple or type(dimensions) == list):
+            raise TypeError('Argument `dimensions` must be of type tuple or list. Got {}'.format(type(dimensions)))
+        if not len(dimensions) > 1:
+            raise ValueError('Argument `dimensions` must have a length of more than 1. Got {}'.format(len(dimensions)))
+        if not all(type(dim) == int for dim in dimensions):
+            raise TypeError('Dimension indices must all be of type int.')
+        if not all(dim >= 0 for dim in dimensions):
+            raise ValueError('Dimension index must be positive')
+        if not (type(value) == int or type(value) == float):
+            raise TypeError('Argument `value` must be of type float or int. Got {}'.format(type(value)))
+
+        self.dimensions = tuple(dimensions)
+        self.value = value
+
+        self.validate_sample = self._validate_sample
+
+    def _validate_sample(self, sample: Sequence[Any]) -> bool:
```

Comment:
I think `Sequence[Any]` is a synonym for `Sequence`, so this can be simplified. If we are to be VERY general, it only uses that `sample` is an iterable (see https://docs.python.org/3/library/collections.abc.html?#collections-abstract-base-classes), this can be generalised to
```suggestion
    def _validate_sample(self, sample: Iterable) -> bool:
```